qgscamerapose: Fix pitch min value to avoid rotation computation error

ptitjano opened the pull request:
This prevents a bug in QgsCameraPose::updateCamera when updating
camera rotation.
Indeed, with a mPitchAngle < 0.2, QQuaternion::fromEulerAngles(
mPitchAngle, mHeadingAngle, 0 ) will return bad rotation angle.

This is fixed is Qt6, but it has not been backported to Qt5.
See: https://bugreports.qt.io/browse/QTBUG-72103

cc @benoitdm-oslandia

benoitdm-oslandia committed:
qgscamerapose: Fix pitch min value to avoid rotation computation error

This prevents a bug in QgsCameraPose::updateCamera when updating
camera rotation.
Indeed, with a mPitchAngle < 0.2, QQuaternion::fromEulerAngles(
mPitchAngle, mHeadingAngle, 0 ) will return bad rotation angle.

This is fixed is Qt6, but it has not been backported to Qt5.
See: https://bugreports.qt.io/browse/QTBUG-72103

ptitjano commented:
@wonder-sk Could you take a look at it please? This should fix an issue mentioned in the [3D axis PR](https://github.com/qgis/QGIS/pull/48109#issuecomment-1145007215).

lbartoletti commented on src/3d/qgscamerapose.cpp:
```diff
@@ -62,8 +62,11 @@ void QgsCameraPose::setPitchAngle( float pitch )
     mPitchAngle = 180.0f;
-  else if ( pitch < 0.0f )
-    mPitchAngle = 0.0f;
+  else if ( pitch < 0.2f )
```
```suggestion
  else if ( pitch < 0.2f )
```

Maybe use `std::clamp` from algorithm include:

`std::clamp(mPitchAngle, 0.2f, 180.0f)`

ptitjano replied:
Good idea. Applied. Thanks

ptitjano committed:
qgscamerapose: Use std::clamp to clamp mPitchAngle value

std::clamp was introduced in c++17. This is a shorter version of the
if/else if/else syntax.

ptitjano committed:
tests: Update 3d rendering tests with new pitch min value

The previous commit set a new min value for the pitch value (0.2
instead of 0) to prevent a Qt3d bug. A lot of the tests set the pitch
value to 0 to export the scene. The expected images are updated to
take this difference into account.

uclaros commented:
@ptitjano This has fixed the 3D axis issue only partially. The axes are now ok when viewing the scene from top, but the issue still appears when the scene is viewed from the bottom.
